xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPIGL.cpp

```diff
@@ -75,9 +75,31 @@ bool CRendererVAAPI::Configure(const VideoPicture &picture, float fps, unsigned
   interop.glEGLImageTargetTexture2DOES = (PFNGLEGLIMAGETARGETTEXTURE2DOESPROC)eglGetProcAddress("glEGLImageTargetTexture2DOES");
   interop.eglDisplay = CRendererVAAPI::m_pWinSystem->GetEGLDisplay();
 
+  bool useVaapi2{};
+  bool general{}, hevc{};
+#if KODI_HAVE_VAAPI2TEXTURE
```

Comment:
remove the ifdef here, instead make CVaapi2Texture::TestInterop return false, if !KODI_HAVE_VAAPI2TEXTURE

Reply:
OK, that would mean ifdef-ing every function in CVaapi2Texture though? Either in the header or in the implementation.

Reply:
you could define a kind of stub class for this case. then no ifdefs are need for the actual class.

Reply:
Mh don't really like that idea, then people will forget to keep the stub in sync and build will break on older libva. I've ifdef-ed the places that need it, there weren't that many after all.

Reply:
I won't forget and I maintain this code